rfcs/039-requesting-api-design/README.md

```diff
@@ -0,0 +1,150 @@
+# Requesting API design
+
+## Background
+
+As part of the work to add requesting, we need to add new some new APIs.
+In particular, these APIs should allow a user to:
+
+-   Get the status of a given item (e.g. available, on hold)
+-   Request an item
+-   See a list of their current requests
+
+Relinquishing a request is out of scope for the initial API.
+There's currently no automated process to do this, and we'd need to coordinate with LE&E to define such a process.
+
+This RFC describes how these APIs will work, and provides examples of requests and responses.
+
+It does *not* suggest implementation details – those can be discussed separately.
+
+
+
+## Guiding principles
+
+1.  **We should not assume requests are sent to a particular service.**
+
+    Although the initial implementation will be making requests for items in Sierra, we may add support for requesting from other systems in a future update.
+    We should not assume the use of Sierra.
+
+2.  **We should use platform identifiers throughout.**
+
+    We may include source identifiers, but interactions with these APIs should start with platform identifiers (e.g. `a222wwjt`).
+    This is a consequence of principle (1).
+
+
+
+## Proposed APIs
+
+### Get the status of a given item
+
+We will add a new endpoint `/catalogue/v2/works/{workId}/items`.
+
+-   This will return a modified version of our existing `Work` model.
+
+    In particular, responses will have only two fields: `id` and `items`.
+    The `Item` model will in turn only have two fields: `id` and `status`, the latter of which returns an instance of a new `ItemStatus` type, e.g.
+
+    ```
+    "status": {
+      "id" : "available",
+      "label" : "Available",
+      "type": "ItemStatus"
+    }
+    ```
+
+-   This will return an item for every identified item in the corresponding `/works` request.
+
+    If an item cannot be requested (e.g. it is tracked in Calm, not Sierra), we will return a `not-requestable` status.
+    This includes items that only have digital locations.
+
+-   We will apply rate limiting to this endpoint, and possibly an API key (tbd).
+
+    Requests to this endpoint will trigger requests to Sierra (and possibly other systems in future).
+    We don't want it to be used to DDOS those systems, so we need a way to turn off misbehaving clients.
+
+<details>
+<summary>Example request/response</summary>
+
+```http
+GET /catalogue/v2/works/a23upgqy/items
+{
+  "id": "a23upgqy",
+  "items": [
+    {
+      "id": "zv4gvvq8",
+      "status": {
+        "id" : "available",
+        "label" : "Available",
+        "type": "ItemStatus"
+      },
+      "type": "Item"
+    },
+    {
+      "id": "fwsbxqp4",
+      "status": {
+        "id" : "not-requestable",
+        "label" : "Not requestable",
+        "type": "ItemStatus"
+      },
+      "type": "Item"
+    }
+  ],
+  "type": "Work",
```

Comment:
I'm not sure about this (and hence also https://github.com/wellcomecollection/docs/pull/56/files#diff-23afd2605893f14d49f2d9d49cc5aa1966d3e72a02bffe5958ef560e17591321R41) - this doesn't have very much in common with a work and from a REST pedantry perspective the URL implies a collection / resource list rather than an atomic resource...

Could it be an `ItemsList` or something?

Reply:
That or we modify our description of a `Work` to imply that all the missing fields are optional. I think @jtweed will have an opinion on this.

Reply:
You mean the missing fields in `items`? hmmmm...

Reply:
Both, I think.

Reply:
Having checked – the only compulsory fields on DisplayItem are id and locations (which can be an empty list), so omitting that doesn't seem too much of a sin. The Work is a bigger issue, I think.

Reply:
`DisplayResultList[DisplayItem]`?

Reply:
although that doesn't give you the id back...